### ISSUE TITLE
outdated h91 technique

### ISSUE THREAD

**Issue by goetsu** [w3c/wcag#167]
Hi,

I think this technique need to be updated to cover at least aria-label and aria-labelledby.
https://www.w3.org/TR/WCAG20-TECHS/H91.html

Regards,

Aurélien

**Comment by goetsu** [comment]
maybe this issue https://github.com/w3c/wcag/issues/168 need to be include in a futur update of this technique too

**Comment by awkawk** [comment]
Technique H91 is focused on the native HTML features that support accessibility.  As ARIA is now allowed in HTML5 I suggest that we add a note above the table in H91 that recognizes that ARIA is referenced in the HTML5 spec and if the native features in HTML are not sufficient that authors may want to consider using ARIA attributes such as aria-label and aria-labelled by, and possibly reference the WCAG ARIA techniques page and/or https://www.w3.org/TR/html5/dom.html#wai-aria

**Comment by karlgroves** [comment]
(I had originally disagreed with @goetsu but am editing)

Per my reading, this is about the control itself and not the labeling of the control. That is covered in H44. 

> The objective of this technique is to use standard HTML form controls and link elements to provide keyboard operation and assistive technology interoperability of interactive user interface elements.

As the sentence above indicates, providing these controls using standard HTML ensures that, at the very least, the controls will have proper roles, states, and properties for that type of control and be keyboard accessible (as long as none of these things are otherwise interfered with).

However, the rest of the content in this technique goes on to discuss labelling of controls and unfortunately shows the use of title attributes.  Given this, I think that the examples provided should include ARIA as suggested by @goetsu

**Comment by goetsu** [comment]
H44 is also outdated in that case ;)

**Comment by awkawk** [comment]
I'm a little concerned about this pulling in more and more of the ARIA spec.  I do agree that we want to mention that ARIA can be used to provide the accessibility name for a control, but since the table is also talking about role and state, do we need to indicate that ARIA can override the role and that the aria states need to be verified also?

I think that the goal of this technique is to show that the native controls address the keyboard and AT support needs of WCAG easily, and the more that ARIA is introduced the more complications and caveats we need to provide.

**Comment by joshueoconnor** [comment]
I also agree with not doing this, as it is a simple HTML technique and does not need ARIA at all.

**Comment by goetsu** [comment]
at least maybe we can  just add a note to suggest ARIA4, ARIA5, ARIA14 and ARIA16 can be used in some context

**Comment by joshueoconnor** [comment]
Working group resolution:
The working group has agreed to add links to related ARIA techniques such as ARIA 14/16.

**Comment by goetsu** [comment]
sound fine for me. The same is needed for H44

**Comment by awkawk** [comment]
Pull request to address this: https://github.com/w3c/wcag/pull/179/files?diff=split

**Comment by jnurthen** [comment]
I have a question reviewing this as to what "value" actually means in this
technique. Can anyone clarify what this is? It certainly isn't the HTML
value atrtibute (href for links doesn't fit this model), it isn't the DOM
value property. What is it referring to?

**Comment by awkawk** [comment]
It is the value as mentioned in 4.1.2:
For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies.

I interpret this to be the content of the element, whether set by default by the author or input by the end-user. For example, the value for a input of type=text is set by the value attribute (which may be empty or missing by default).  For an combo box the value comes from a selected option element.  

In my edits I also used value to indicate what the content of an attribute is, as in "The alt attribute value is used to provide the accessibility name for an input of type=image".

Does this help?

**Comment by jnurthen** [comment]
I guess it is really the one for links I am struggling with then. I don't
see how the href is really the "value" of a link. In the definition of
4.1.2 value only comes up in terms of things that "can be set by the user".
The href of a link cannot be set by the user.

**Comment by awkawk** [comment]
I was wondering about that one as well.  We should check what the accessibility API's regard as the value for a link.

**Comment by mraccess77** [comment]
The button element also seems to be labellable by the label element.  Seems to appear in the API correctly for browsers -- but not sure what the AT support is for this technique.

**Comment by alastc** [comment]
It appears the changes are still valid, but not live therefore need re-applying in the HTML version of H91?

**Comment by fstrr** [comment]
This issue was originally opened over six years ago and last updated nearly three years ago. Does this still need attention or can it be closed?

**Comment by goetsu** [comment]
@alastc @awkawk this issue as been close without the modification being done can it be reopened ?

**Comment by patrickhlauke** [comment]
techniques aren't comprehensive, they only show one way to satisfy the requirements of a criterion. there may be other ways that aren't documented. so the fact that there may be more ways, like using `aria-label` / `aria-labelledby`, that aren't also included in the technique is ... fine/not a problem, as auditors and authors need to refer back to the normative requirement of the SCs, not just go by existing techniques...

if you want to take a stab at making a PR to update this though, go for it

**Comment by goetsu** [comment]
@patrickhlauke https://github.com/w3c/wcag/pull/4975
